Archlinux

VileLasagna opened the pull request:
Initial porting to Arch Linux (implements #10 ). Current state is:
 - Src install is apparently working fine
 - Packaging is disabled, I've introduced exit code 2 to the scripts to represent a deliberate abort
   - Code is still there commented out to serve as guide when sorting this out
 - READMEs updated
 - DKMS related files gone. Given how fast Arch rolls, they're likely to be unnecessary anyway
 
 - Most of the changes are simply disabling packaging and "correcting" package manager calls
  - There are, however, three specific workarounds: 
    - `03_00_rocblas.sh` has a sed call to avoid wrong python executable being selected. There is a comment mentioning a PR, as of the opening of THIS PR, that one has actually been merged into their develop branch and should be in on the next release
    - `03_01_hipblas.sh` has a different fix that prevents the build from failing on its own if CUDA is at all installed (it tries to enable it if it finds it, regardless of any options you are able to provide, but can't build for CUDA and HIP at the same time)
    - `01_07_opencl.sh` has another python2 related fix, this time merely explicitly calling it instead of "standard python interpreter`

VileLasagna committed:
Initial conversion of source build scripts to ArchLinux

haasanen committed:
Fix ATMI build on Arch Linux: add rsync to pre-reqs for ATMI build

VileLasagna committed:
Updated README.MD for Arch Linux

VileLasagna committed:
Moved arch scripts to src_install subfolder

VileLasagna committed:
Renamed a couple of scripts in Arch src_install to match conventions in other distros

VileLasagna committed:
Added a README.MD for arch/src_install

VileLasagna committed:
Removed DKMS module related files from Arch scripts

VileLasagna committed:
Provided an useful message and exit for non-supported packaging in Arch

VileLasagna committed:
Corrected symlinks for patches

VileLasagna committed:
Reverted some thread allocation logic to the original conservative values

jlgreathouse commented:
Hi @VileLasagna 

Thanks for submitting this. I won't be able to get around to testing / reviewing this until next week due to some other high priority work on my plate right now. However, I wanted to note that I've seen this and I'm thankful for your help!

hameerabbasi commented:
Hello, I can help test, but probably not review. I'm getting the following error:

```
./00_prepare_system_arch.sh: line 28: parse_args: command not found
"parse_args "$@"" command failed with exit code 127.
```

I'm assuming you forgot some package... I actually did install `argparser` via AUR but it doesn't seem to work.

VileLasagna commented:
This is a bit weird. I'm pretty sure parse_args should be a bash command itself? I'll nevertheless have a look later today

hameerabbasi commented:
I thought the same at first... But a simple Google search for "parse_args bash" reveals that's not the case. Is this what you're looking for? https://github.com/flyingangel/argparser or https://aur.archlinux.org/packages/argparser-bash/

VileLasagna commented:
Hello, man. Sorry I didn't get back to you earlier. Accidentally marked the email as read and this slip my mind.

So, the parse_args function is defined in common_options which is sourced.
the way this happens is:

```
BASE_DIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
.
.
.
source "$BASE_DIR/common_options.sh"
parse_args "$@"
```

I've never used that specific way of getting the current directory so maybe it's breaking if you're calling from a different directory? This I didn't touch from the original files but if it proves annoying, it's easy enough to look more closely

hameerabbasi commented:
With `master` the original issue has been resolved but I face this error now:

```
Cloning into 'rocminfo'...
remote: Enumerating objects: 4, done.
remote: Counting objects: 100% (4/4), done.
remote: Compressing objects: 100% (4/4), done.
remote: Total 72 (delta 0), reused 1 (delta 0), pack-reused 68
Unpacking objects: 100% (72/72), done.
Note: checking out '1bb0ccc'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 1bb0ccc Add hex representation of some values to output
-- The C compiler identification is GNU 8.2.1
-- The CXX compiler identification is GNU 8.2.1
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Error at CMakeLists.txt:18 (message):
  Can't find hsa.h.


-- Configuring incomplete, errors occurred!
See also "/tmp/tmp.X97S0nYRWa/rocminfo/build/CMakeFiles/CMakeOutput.log".
"cmake .. -DCMAKE_BUILD_TYPE=${ROCM_CMAKE_BUILD_TYPE} -DCMAKE_INSTALL_PREFIX=${ROCM_OUTPUT_DIR}/ -DCPACK_PACKAGING_INSTALL_PREFIX=${ROCM_OUTPUT_DIR}/ -DCPACK_GENERATOR=DEB" command failed with exit code 1.
"${BASE_DIR}/component_scripts/01_05_rocminfo.sh "$@"" command failed with exit code 1.
```

I'm sorry for taking so long, again.

VileLasagna commented:
You say master but that is curious as this hasn't been merged yet...

Not being able to find the header would indicate that either you aren't installing the packages as you go or that you're deploying them to a different folder and haven't passed the correct "output and input" folders for your local deployment.

Sry I haven't been able to look at this more closely, I'm a bit sidetracked with another project for a bit, but do plan on coming back for further changes.

Of note: Both of my pull requests in rocblas and hipblas have been accepted so once the version of those libraries roll over I can drop the workarounds on this side

VileLasagna committed:
Copied over files Fedora scripts as base for OpenSUSE

VileLasagna committed:
Changed package manager calls to Zypper

VileLasagna committed:
Fixed source build for ROCm install in OpenSUSE 15

VileLasagna committed:
[OpenSuSE] Applied python fix for rocblas

VileLasagna committed:
[OpenSuSE] Applied CUDA workaround in hipBLAS build

VileLasagna committed:
[OpenSuSE] Added 03_04_rocprim.sh. Corrected some package installs

VileLasagna committed:
[OpenSuSE] Added ROCK driver to src build

VileLasagna committed:
[OpenSuSE] Corrected files for RPM packaging in rocm stack(01)

VileLasagna committed:
[OpenSuSE] Corrected scripts fort rocBlas and rocfft

VileLasagna committed:
[OpenSuSE] removed rpm_install folder

VileLasagna committed:
[OpenSuSE] Added a CUDA and CMake version check on prepare system

CMake < 3.13 will not correctly find CUDA >= 10. Since many of the
projects have CUDA sides and/or options, their build scriptscan and will
fail if CUDA is only "partially found"

VileLasagna committed:
[OpenSuSE] Added newer CMake to prepare_system_opensuse

VileLasagna committed:
[OpenSuSE] Added missing dependency to 03_09_miopen.sh

VileLasagna committed:
[Arch] Fixed an issue where rocSPARSE wouldn't build any more

VileLasagna committed:
[Arch] Updated src_install/README.MD to include rocPRIM

VileLasagna committed:
[OpenSUSE] Updated READMEs to include OpenSUSE Leap 15

lucianolorenti commented:
@VileLasagna  Thank you for your great job!

I was able to build all the stack and the python3 tensorflow package with the current state of this branch.

I had to make two changes though, besides removing python2-z3 from ` 01_07_opencl.sh`

- upgrade  ROCM_HIP_CHECKOUT to tags/roc-2.5.0. Firstly, I compiled  everything as is, but then I had to upgrade ROCM_HIP and recompile the packages listed here https://github.com/ROCmSoftwarePlatform/tensorflow-upstream/issues/357#issuecomment-473539412 because I was having errors with `hipSymbolName` (I really don't know if the recompilation of those packages were necessary)

- Finally, I had to force libmiopengemm to link with opencl because it was not happening and at the moment of loading all the libraries related to the GPU inside python, some undefined symbols arose.

VileLasagna commented:
Hey, @lucianolorenti 
Thx.

Curious about the OpenCL stuff. I'll try and have a look in the near(ish) future and see if I can inject some workaround.

As for versions, in both additional distros that I've added support for, I've elected to not touch the versions, as they come from a common file to centralise everything. I'm expecting that with the release of Navi in the near future, this project will need and, hopefully get, a version bump on master and, when it does, I'll update the branch. Until then, I'm avoiding diverging as much as I can

VileLasagna committed:
Revert "[OpenSuSE] Corrected files for RPM packaging in rocm stack(01)"

This reverts commit 31e5b37195078f6ee657414c4d5f8a77735ee168.

VileLasagna committed:
Revert "[OpenSuSE] Corrected scripts fort rocBlas and rocfft"

This reverts commit 848a3c4296fc7218b18cc2a13d8a8df85f971316.

VileLasagna committed:
Revert "[OpenSuSE] Added missing dependency to 03_09_miopen.sh"

This reverts commit 000396985ad78844ea4d5b3042211e4b54cabb43.

VileLasagna committed:
Revert "[OpenSuSE] Applied CUDA workaround in hipBLAS build"

This reverts commit e9de9a82c8862450c6f5d125c5594a2b04056718.

VileLasagna committed:
Revert "[OpenSuSE] Added newer CMake to prepare_system_opensuse"

This reverts commit 498ca478fc960bea937dbeebc282b77c932f6559.

VileLasagna committed:
Revert "[OpenSuSE] removed rpm_install folder"

This reverts commit af90fff0deb49421cae858743109f2aa42dc9527.

VileLasagna committed:
Revert "[OpenSuSE] Added ROCK driver to src build"

This reverts commit 7ea9e4606e5dedef2a856fd31fd20c753c069794.

VileLasagna committed:
Revert "Changed package manager calls to Zypper"

This reverts commit ba96c64fd0f1f36b78df02e8bce210462bac5a35.

VileLasagna committed:
Revert "[OpenSuSE] Applied python fix for rocblas"

This reverts commit 2c04b49c94c1274bc4412391485d79cb97d3ee4b.

VileLasagna committed:
Revert "Copied over files Fedora scripts as base for OpenSUSE"

This reverts commit 2eda4dd2a25528e67ccb86738085e8ada1de2d5e.

VileLasagna committed:
Revert "[OpenSuSE] Added a CUDA and CMake version check on prepare system"

This reverts commit 668678aaf288eabe7797a9473faf241d677d4a5b.

VileLasagna committed:
Revert "[OpenSUSE] Updated READMEs to include OpenSUSE Leap 15"

This reverts commit c75eeb20fcee20089b4d4a1976e1f9d28053038b.

VileLasagna committed:
Revert "[OpenSuSE] Added 03_04_rocprim.sh. Corrected some package installs"

This reverts commit ed61721b48d3790db9cfa77f1e7e4a4a86d2881c.

VileLasagna committed:
Revert "Fixed source build for ROCm install in OpenSUSE 15"

This reverts commit ffef9c6971d49f650b501791b007119a9c9f5857.
